lumenlearning/lti-middleware
# DL3-35 secure middleware courses endpoint

## Description
Secured the middleware's backend endpoint for retrieving the Harmony course catalog using the LTI id_token.

### Motivation and Context
Only users who have a valid LTI registration who are accessing the Deep Linking menu through a valid LTI flow should be permitted to access the course catalog via the middleware.

### Fixes
[DL3-35](https://lumenlearning.atlassian.net/browse/DL3-35)

## How Has This Been Tested?
0. Ensure as a prerequisite that the tool was registered and deployed in D2L and that a Deep Linking Quicklink Link has been added to the Deployment and that there is an org unit appropriately given permission to access the tool.
1. Go to the course (org unit) that has permission to access the tool in D2L.
2. Click on Content in the top navigation.
3. Click "Add a module...", write a name for it, and hit enter to create the module.
4. Click the dropdown for Existing Activities
5. Click on the Deep Linking Quicklink (e.g. I named mine "Lumen Content Selector").
6. Ensure that the course catalog displays with the expected data being sent from the Harmony /course_catalog endpoint and that the pagination is in place if there are more than 10 courses.
Note: If all of the courses in the catalog are using the brain image and were released on 1/18/2022, then this is an indication that the call to Harmony failed and test data has been returned.

## Screenshots
N/A
---
## Database changes
N/A

## ⚠️ Deployment instructions ⚠️
N/A

## New ENV variables
N/A

---

## Checklist
